x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/flyout_components/add_to_rules_table/use_add_to_rules_table.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { sortBy } from 'lodash';
+import type {
+  CriteriaWithPagination,
+  EuiBasicTableColumn,
+  HorizontalAlignment,
+} from '@elastic/eui';
+import { EuiLoadingContent } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import * as myI18n from './translations';
+
+import { useFindRulesInMemory } from '../../../../rule_management_ui/components/rules_table/rules_table/use_find_rules_in_memory';
+import type { Rule } from '../../../../rule_management/logic/types';
+import { getRulesTableColumn } from '../utils';
+import { LinkRuleSwitch } from './link_rule_switch';
+
+export interface ExceptionsAddToRulesComponentProps {
+  initiallySelectedRules?: Rule[];
+  onRuleSelectionChange?: (rulesSelectedToAdd: Rule[]) => void;
+}
+export const useAddToRulesTable = ({
+  initiallySelectedRules,
+  onRuleSelectionChange,
+}: ExceptionsAddToRulesComponentProps) => {
+  const { data: { rules } = { rules: [], total: 0 }, isFetched } = useFindRulesInMemory({
+    isInMemorySorting: true,
+    filterOptions: {
+      filter: '',
+      showCustomRules: false,
+      showElasticRules: false,
+      tags: [],
+    },
+    sortingOptions: undefined,
+    pagination: undefined,
+    refetchInterval: false,
+  });
+
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    initialPageSize: 5,
+    showPerPageOptions: false,
+  });
+
+  const [linkedRules, setLinkedRules] = useState<Rule[]>(initiallySelectedRules || []);
+  useEffect(() => {
+    if (typeof onRuleSelectionChange === 'function') onRuleSelectionChange(linkedRules);
```

Comment:
nit: i see this is typed as function | undefined, but all uses of it seem to be defined as a function, could probably change the type and remove this if

Reply:
please check https://github.com/elastic/kibana/pull/146121/commits/6b7f2e787763152a33415b07fcf42907e591a1f0